src/components/Payment.js

```diff
@@ -19,6 +24,15 @@ class Payment extends Component {
     this.setState({ balance: this.props.startingBalance - paid });
   }
 
+  setCreditTip(e) {
+    this.setState({ creditTip: e.target.value });
+  }
+
+  allInputAmounts() {
+    const amounts = this.state.amountInputs.map((input) => input.value).concat(this.state.creditTip);
+    return amounts;
+  }
+
   componentDidMount() {
     const amountInputs = Array.from(this.refs.form.querySelectorAll('.payment-amount-input'));
```

Comment:
the implementation of this would also have to change if we were to do an amount object - { cash, credit, creditTip }